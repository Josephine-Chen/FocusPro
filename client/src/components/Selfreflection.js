import React from 'react';
import { BootstrapTable, TableHeaderColumn } from 'react-bootstrap-table';
import moment from 'moment';
import { getReflectionsAjax } from '../utils/SettingsUtil';

export default class Selfreflection extends React.Component {
  constructor(props) { // Hand downs the prop chain
    super(props);
    this.state = {
      profile: this.props.auth.getProfile(), // it gets imported here
      reflections: [],
      answer: '',
      question: '',
    };
    this.getReflections = this.getReflections.bind(this);
    this.handleCAnswerhange = this.handleAnswerChange.bind(this);

    this.options = {
      defaultSortName: 'date',  // default sort column
      defaultSortOrder: 'asc',  // default sort order
    };
  }

  componentDidMount() {
    this.getReflections();
    this.callCustomJQuery();
  }

  getReflections() {
    const that = this;
    getReflectionsAjax(this.state.profile.user_id, (data) => {
      const reflectionData = data.data.slice();
      // transform data into format accepted by datatable
      (data.data).forEach((refl, index) => {
        reflectionData[index].date = moment(refl.updatedAt).calendar();
      });
      that.setState({
        reflections: reflectionData,
      });
    });
  }

  getCaret(direction) {
    if (direction === 'asc') {
      return ( <i className="fa fa-caret-up" aria-hidden="true" /> );
    }
    if (direction === 'desc') {
      return (
        <i className="fa fa-caret-down" aria-hidden="true" />
      );
    }
    return (
      <i className="fa fa-sort" aria-hidden="true" />
    );
  }

  handleAnswerChange(event) {
    this.setState({
      answer: event.target.value
    });
  }

  callCustomJQuery() {
    $('.collapsible').collapsible();
  }

  render() {
    return (
      <div>
        <h1> Self-Reflection </h1>

        <BootstrapTable data={this.state.reflections} options={this.options} striped hover pagination>
          <TableHeaderColumn hidden isKey dataField="id">ID</TableHeaderColumn>
          <TableHeaderColumn dataSort dataField="question" tdStyle={{ whiteSpace: 'normal' }} caretRender={this.getCaret}>Question </TableHeaderColumn>
          <TableHeaderColumn dataField="answer" tdStyle={{ whiteSpace: 'normal' }}>Answer </TableHeaderColumn>
          <TableHeaderColumn dataSort dataField="date" tdStyle={{ whiteSpace: 'normal' }} caretRender={this.getCaret} width="100">Date </TableHeaderColumn>
        </BootstrapTable>
      </div>
    );
  }
}

