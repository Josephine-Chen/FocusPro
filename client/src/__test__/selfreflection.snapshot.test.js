import React from 'react';
import renderer from 'react-test-renderer';

//children mock need to be defined before parent module import
jest.mock('react-bootstrap-table', () => {
  return {
    BootstrapTable: 'BootstrapTable',
    TableHeaderColumn: 'TableHeaderColumn',
  };
});

jest.mock('../utils/SettingsUtil', () => {
  const dummyQues ={
    data: [{
            question: 'What help can you get?',
            answer: 'Get a couch to hold myself accountable',
            updatedAt: '2017-03-05T00:13:39.289Z'
          }, {
            question: 'Whats the worst thing that could happen?',
            answer: 'nothing',
            updatedAt: '2017-03-05T23:19:16.943Z'
    }],
  };
  return {
    getReflectionsAjax: (userId, callback) => {
      callback(dummyQues);
    },
  }
});

import Selfreflection from '../components/Selfreflection.js';

describe('Selfreflection (Snapshot)', () => {
  beforeAll(() => {
    global.auth = {
      getProfile: ()=>'dummy',
      'event': {
        on : ()=>'dummy event listener'
      }
    };
  });

  afterAll(() => {
    global.auth = null;
  });

  it('Selfreflection renders', () => {
    const component = renderer.create(<Selfreflection auth={auth}/>);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});