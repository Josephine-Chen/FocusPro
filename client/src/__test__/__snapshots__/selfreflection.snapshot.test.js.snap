exports[`Selfreflection (Snapshot) Selfreflection renders 1`] = `
<div>
  <h1>
     Self-Reflection 
  </h1>
  <BootstrapTable
    data={
      Array [
        Object {
          "answer": "Get a couch to hold myself accountable",
          "date": "Yesterday at 4:13 PM",
          "question": "What help can you get?",
          "updatedAt": "2017-03-05T00:13:39.289Z",
        },
        Object {
          "answer": "nothing",
          "date": "Today at 3:19 PM",
          "question": "Whats the worst thing that could happen?",
          "updatedAt": "2017-03-05T23:19:16.943Z",
        },
      ]
    }
    hover={true}
    options={
      Object {
        "defaultSortName": "date",
        "defaultSortOrder": "asc",
      }
    }
    pagination={true}
    striped={true}>
    <TableHeaderColumn
      dataField="id"
      hidden={true}
      isKey={true}>
      ID
    </TableHeaderColumn>
    <TableHeaderColumn
      caretRender={[Function]}
      dataField="question"
      dataSort={true}
      tdStyle={
        Object {
          "whiteSpace": "normal",
        }
      }>
      Question 
    </TableHeaderColumn>
    <TableHeaderColumn
      dataField="answer"
      tdStyle={
        Object {
          "whiteSpace": "normal",
        }
      }>
      Answer 
    </TableHeaderColumn>
    <TableHeaderColumn
      caretRender={[Function]}
      dataField="date"
      dataSort={true}
      tdStyle={
        Object {
          "whiteSpace": "normal",
        }
      }
      width="100">
      Date 
    </TableHeaderColumn>
  </BootstrapTable>
</div>
`;
